Modix/Modules/MarkerRoleModule.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Microsoft.AspNetCore.Mvc.TagHelpers;
+
+namespace Modix.Modules
+{
+    [Group("pingrole"), Name("Marker Role Manager"), Summary("Allows you to add and remove specific marker roles.")]
+    public class MarkerRoleModule : ModuleBase
+    {
+        private List<ulong> _roleIds;
+
+        public MarkerRoleModule(List<ulong> roleIds)
+        {
+            _roleIds = roleIds;
+        }
+
+        [Command("add")]
+        public async Task AddRole([Remainder] IRole targetRole) 
+        {
+            if (!_roleIds.Contains(targetRole.Id))
+            {
+                await ReplyAsync("Cannot add that role.");
+                return;
+            }
+
+            var guildUser = (IGuildUser) Context.User;
+            await guildUser.AddRoleAsync(targetRole);
+            await ReplyAsync($"Added role `{targetRole.Name}`.");
+        }
+
+        [Command("remove")]
+        public async Task RemoveRole([Remainder] IRole targetRole)
+        {
+            if (!_roleIds.Contains(targetRole.Id))
+            {
+                await ReplyAsync("Cannot remove that role.");
+                return;
+            }
+
+            var guildUser = (IGuildUser) Context.User;
+            await guildUser.RemoveRoleAsync(targetRole);
+            await ReplyAsync($"Removed role `{targetRole.Name}`.");
+        }
+
+        [RequireUserPermission(GuildPermission.ManageRoles)]
```

Comment:
Now that we have `IAuthorizationService` in place you should maybe use that instead. This would also probably require pushing all the business logic here down into an `IMarkerRoleService`, so it can all be reused by the web app if we want to make a frontend page there for managing the roles.

No need for this to be done right now.